### ISSUE TITLE
Mac x86_64 wrong shape of output 

### ISSUE THREAD

**Issue by pinguo-huangliang** [Tencent/ncnn#3638]
## error log | 日志或报错信息 | ログ
wrong output shape of convolution
Mac x86_64 机器上 卷积层 创建 输出blob shape 不对
（一部分信息来自.param，一部分信息来自bottom_blob，但bottom_blob 的packing模式会根据情况改变）

## context | 编译/运行环境 | バックグラウンド
Mac x86_64 pytorch onnx
## how to reproduce | 复现步骤 | 再現方法
1. create pytorch model and export to onnx( test_module.onnx) :    
class test_module(nn.Module):
    def __init__(self):
        super(test_module, self).__init__()
        self.conv_layer = torch.nn.Conv2d(in_channels=8,
                                          out_channels=8,
                                          kernel_size=1, stride=1, padding=0, groups=1, bias=False)
    def forward(self, x):
        x = self.conv_layer(x)
        return x

3. convert to ncnn model:
onnx2ncnn test_module.onnx test_module.param test_module.bin

5. run c++ code:
        ncnn::Net net;
        net.load_param("test_module.param");
        net.load_model("test_module.bin");

        ncnn::Extractor ex = net.create_extractor();
        ex.set_num_threads(1);

        int w = 256;
        int h = 256;
        int d = 1;
        int c = 8;
        ncnn::Mat input(w,h,d,c,sizeof(float));
        ncnn::Mat output;

        ex.input("input", input);
        ex.extract("output", output);

        std::cout<<"c = "<<output.c<<std::endl;
        std::cout<<"d = "<<output.d<<std::endl;
        std::cout<<"h = "<<output.h<<std::endl;
        std::cout<<"w = "<<output.w<<std::endl;
        std::cout<<"elemsize = "<<output.elemsize<<std::endl;
        std::cout<<"elempack = "<<output.elempack<<std::endl;
        
6. got result:
c = 8
d = 1
h = 256
w = 256
elemsize = 32
elempack = 1

I think 'elemsize' should be 4 (sizeof(float));

7: my opinion
I looked into the source code of ncnn,and find that the problem is code bug of packing:

a. when input mat with shape (c,d,h,w) , elemsize = sizeof(float) , elempack = 1, ncnn will pack it , and make it a new view : (c/8,d,h,w) , elemsize = sizeof(float)*8 , elempack = 8, this happens in 'net.cpp->NetPrivate::do_forward_layer->convert_layout(bottom_blob, layer, opt);'

b.in convolution (Convolution.cpp->Convolution::forward),
![截屏2022-03-19 下午5 33 22](https://user-images.githubusercontent.com/9360214/159115943-783a0ae3-d3e2-4ac6-998b-8caedbb3a15b.jpg)

 top_blob.create(outw, outh, num_output, elemsize, opt.blob_allocator);

outw, outh, num_output  are from 'test_module.param' file, but elemsize is from 'bottom_blob_bordered', 
here,  I think ncnn should use 'num_output/(bottom_blob_bordered.elempack)' instead of 'num_output'

**Comment by nihui** [comment]
输入的 ```ncnn::Mat input``` 应当是3维shape，即 ```ncnn::Mat input(w,h,c)```

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
